tracer/src/Datadog.Trace/AppSec/Coordinator/SecurityCoordinator.Core.cs

```diff
@@ -166,7 +166,27 @@ internal override bool IsBlocked
             }
         }
 
-        internal override int StatusCode => Context.Response.StatusCode;
+        internal override int? StatusCode
+        {
+            get
+            {
+                if (IsAdditiveContextDisposed())
+                {
+                    return null;
+                }
+
+                try
+                {
+                    return Context.Response.StatusCode;
+                }
+                catch (Exception e) when (e is NullReferenceException or ObjectDisposedException)
+                {
+                    Log.Debug(e, "Exception while trying to access StatusCode of a Context.Response.");
```

Comment:
As this is on the hot path, I think this would be good:
```suggestion
                    if (Log.IsEnabled(LogEventLevel.Debug))
                    {
                        Log.Debug(e, "Exception while trying to access StatusCode of a Context.Response.");
                    }
```